evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -150,6 +164,41 @@ fn prank(
     Ok(Bytes::new())
 }
 
+fn read_callers(state: &Cheatcodes, default_sender: Address) -> Bytes {
+    let Cheatcodes { prank, broadcast, .. } = &state;
+
+    let data = if let Some(prank) = prank {
+        let caller_mode =
+            if prank.single_call { CallerMode::Prank } else { CallerMode::RecurrentPrank };
+
+        [
+            Token::Uint(caller_mode.into()),
+            Token::Address(prank.new_caller),
+            Token::Address(prank.new_origin.unwrap_or(default_sender)),
+        ]
+    } else if let Some(broadcast) = broadcast {
```

Comment:
I think it would be good to document why does behavior differ on broadcast. It's a bit diffuse due to all the conversation on the PR 😅

Reply:
Done 👌🏼